ts/defs.bzl

```diff
@@ -68,7 +72,169 @@ def ts_project(
         **kwargs):
     """Compiles one TypeScript project using `tsc --project`.
 
-    For the list of args, see the ts_project rule.
+    This is a drop-in replacement for the `tsc` rule automatically generated for the "typescript"
+    package, typically loaded from `@npm//typescript:package_json.bzl`.
+    Unlike bare `tsc`, this rule understands the Bazel interop mechanism (Providers)
+    so that this rule works with others that produce or consume TypeScript typings (`.d.ts` files).
+
+    Some TypeScript options affect which files are emitted, and Bazel needs to predict these ahead-of-time.
+    As a result, several options from the tsconfig file must be mirrored as attributes to ts_project.
+    A validator action is run to help ensure that these are correctly mirrored.
+    See https://www.typescriptlang.org/tsconfig for a listing of the TypeScript options.
+
+    Any code that works with `tsc` should work with `ts_project` with a few caveats:
+
+    - ts_project` always produces some output files, or else Bazel would never run it.
+      Therefore you shouldn't use it with TypeScript's `noEmit` option.
+      If you only want to test that the code typechecks, instead use
+      ```
+      load("@npm//typescript:package_json.bzl", "bin")
+      bin.tsc_test( ... )
+      ```
+    - Bazel requires that the `outDir` (and `declarationDir`) be set beneath
+      `bazel-out/[target architecture]/bin/path/to/package`
+      so we override whatever settings appear in your tsconfig.
+    - Bazel expects that each output is produced by a single rule.
+      Thus if you have two `ts_project` rules with overlapping sources (the same `.ts` file
+      appears in more than one) then you get an error about conflicting `.js` output
+      files if you try to build both together.
+      Worse, if you build them separately then the output directory will contain whichever
+      one you happened to build most recently. This is highly discouraged.
+
+    Args:
+        name: a name for this target
+
+        srcs: List of labels of TypeScript source files to be provided to the compiler.
+            If absent, the default is set as follows:
+            - Include `**/*.ts[x]` (all TypeScript files in the package).
```

Comment:
Should there be line breaks between the - If * statements? (see the generated output with no `<br />`s

Reply:
sure - though I mostly only care how it will appear when rendered on our docsite :)

Reply:
I think this is just a stardoc bug, it doesn't understand bullet lists indented this way.
https://github.com/bazelbuild/bazel/pull/13403
I fixed it on our fork of stardoc.